app/controllers/application_controller.rb

```diff
@@ -5,6 +5,159 @@ class ApplicationController < Sinatra::Base
+    else
+      redirect to '/login' #if not, have them log in
+    end
+  end
+
+  # CREATE
+  get '/tweets/new' do #goes to create page/form
+    if is_logged_in?
+      erb :'tweets/create_tweet' #allow them to create a tweet if they are logged in
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets' do #takes params from the form
+    if params[:content] == '' #no blank tweets allowed. descriptive error message would be nice
+      redirect to 'tweets/new'
+    else
+      user = User.find_by_id(session[:user_id])
+      @tweet = Tweet.create(content: params[:content], user_id: user.id) #make that tweet
+      redirect to "/tweets/#{@tweet.id}" #don't use '' for string interpolation
+    end
+  end
+
+  # READ
+  get '/tweets/:id' do  #get tweets/id should come after get tweets/new
+    if is_logged_in?
+      @tweet = Tweet.find_by_id(params[:id])   #if logged in, find the tweet & show the tweet
+      erb :'tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  # UPDATE / EDIT
+    get '/tweets/:id/edit' do  #edit comes after  get tweets/id
+    if is_logged_in?
+      @tweet = Tweet.find_by_id(params[:id])  #find the tweet
+      if @tweet.user_id == session[:user_id] #check that the logged in user is the tweet owner
+       erb :'tweets/edit_tweet'   #then allow them to edit
+      else
+        redirect to '/tweets'  #then take back to user's tweets
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  patch '/tweets/:id' do
+    if params[:content] == '' #don't allow them update to a blank tweet
+      redirect to "/tweets/#{params[:id]}/edit"
+    else
+      @tweet = Tweet.find_by_id(params[:id]) #get the tweet
+      @tweet.content = params[:content]  #change the content to params
+      @tweet.save  #save the tweet to the db
+      redirect to "/tweets/#{@tweet.id}" #show the tweet
+    end
+  end
+
+  # DELETE
+  delete '/tweets/:id/delete' do
+    if is_logged_in?
+      @tweet = Tweet.find_by_id(params[:id])  #find that tweet
+      if @tweet.user_id == session[:user_id] #check that the person is the creator
+        @tweet.delete #destroy would work as well?
+        redirect to '/tweets'  #back to main user page
+      else
+        redirect to '/tweets' #if they are logged in but not their tweet, they can't delete
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  # See individual user's tweets don't have to be logged in
+ get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    erb :'users/show'
+  end
+
+  ################ USER ###############
+  # User Sign up
+  get '/signup' do #Displays user sign up page unless person is already logged in
+    if is_logged_in?
+      redirect to '/tweets'
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do #processes the signup form submission
+    if params[:username] == "" || params[:email] == "" || params[:password] == "" # no empties allowed
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.